packages/http/src/validator/validators/body.ts

```diff
@@ -19,19 +23,33 @@ export class HttpBodyValidator implements IHttpValidator<any, IMediaTypeContent>
       return [];
     }
 
-    const encodingStyle = get(content, 'encodings[0].style');
+    if (mediaType && typeIs.is(mediaType, 'application/x-www-form-urlencoded')) {
+      const shallowParsedTarget = shallowParseUrlEncoded(target);
+      target = deepParseUrlEncoded(target);
 
-    if (encodingStyle) {
-      const deserializer = body.get(encodingStyle);
-      if (deserializer && deserializer.supports(encodingStyle) && schema.properties) {
-        const propertySchema = Object.keys(schema.properties)[0];
-        const deserializedObject = deserializer.deserialize(propertySchema, target, Object.values(
-          schema.properties,
-        )[0] as any);
+      const encodings = get(content, 'encodings', []);
+      for (const encoding of encodings) {
+        const allowReserved = get(encoding, 'allowReserved', false);
+        const property = encoding.property;
+        const value = shallowParsedTarget[property];
```

Comment:
What does `shallow` specifically mean here?

Reply:
I had trouble naming that.

Basically it should mean that URL is split by `&` and the key-val pairs are extracted. But percent-encoded entities hadn't been decoded yet.

We need it in that form because:
1) we need key-val pairs in order to match key names with relevant encoding definitions
2) we need before-percent-decoding values in order to look for forbidden characters

Reply:
Path and queryParams?

Reply:
Quite unsure what do you mean by path and queryParams. We are dealing here with body payload.